src/Microsoft.Health.Fhir.Api/Features/Filters/FhirRequestContextResourceTypeFilterAttribute.cs

```diff
@@ -0,0 +1,43 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public class FhirRequestContextResourceTypeFilterAttribute : ActionFilterAttribute
```

Comment:
We might want to think about combining this with `FhirRequestContextRouteNameFilterAttribute.cs` and updating the name to be something about hydrating the `FhirRequestContext`.